### PR TITLE
[EBPF] gpu: compute utilization based on active GPU

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

This PR enables multi-gpu support in the utilization calculations.

### Motivation

Up until now, the calculation defaulted to the first GPU device found. After multi-GPU support has been added, we can rely on the actual device.

### Describe how you validated your changes

Unit test included.

### CODE REVIEW

File: pkg/gpu/aggregator.go [hunk 1]
```diff
@@ -38,13 +38,13 @@ type aggregator struct {
 	// processTerminated is true if the process has ended and this aggregator should be deleted
 	processTerminated bool
 
-	// sysCtx is the system context with global GPU-system data
-	sysCtx *systemContext
+	// deviceMaxThreads is the maximum number of threads the GPU can run in parallel, for utilization calculations
```
Comment (val06) [inline pkg/gpu/aggregator.go:41]: is this parameter is the same for ALL gpu devices available on specific host?

File: pkg/gpu/stats.go [hunk 2]
```diff
@@ -84,7 +87,19 @@ func (g *statsGenerator) getOrCreateAggregator(sKey streamKey) *aggregator {
 	}
 
 	if _, ok := g.aggregators[aggKey]; !ok {
-		g.aggregators[aggKey] = newAggregator(g.sysCtx)
+		gpuDevice, err := g.sysCtx.getDeviceByUUID(sKey.gpuUUID)
+		if err != nil {
+			log.Warnf("Error getting device by UUID %s: %s, defaulting to device 0", sKey.gpuUUID, err)
+			gpuDevice = g.sysCtx.gpuDevices[0]
```
Comment (val06) [inline pkg/gpu/stats.go:93]: do we really want that logic? this can mask potential metrics in-accuracies

File: pkg/gpu/stats.go [hunk 3]
```diff
@@ -84,7 +87,19 @@ func (g *statsGenerator) getOrCreateAggregator(sKey streamKey) *aggregator {
...
+		}
+
+		maxThreads, ret := gpuDevice.GetNumGpuCores()
+		if ret != nvml.SUCCESS {
+			log.Warnf("Error getting number of GPU cores: %s, defaulting to 1", nvml.ErrorString(ret))
+			maxThreads = 1
+		}
```
Comment (val06) [inline pkg/gpu/stats.go:100]: same question here

**Review by val06 — COMMENTED** [review]
few questions about implementation

File: pkg/gpu/aggregator.go (hunk 1 above)
Comment (gjulianm) [reply pkg/gpu/aggregator.go:41]: No, but each aggregator only deals with a single device. This way we remove that dependency on the `systemContext`.

File: pkg/gpu/stats.go (hunk 2 above)
Comment (gjulianm) [reply pkg/gpu/stats.go:93]: You're right that this probably would be better if we returned an error.

Comment (gjulianm) [reply pkg/gpu/stats.go:93]: Changed to return an error.

File: pkg/gpu/stats.go (hunk 3 above)
Comment (gjulianm) [reply pkg/gpu/stats.go:100]: Changed to return an error.

**Review by val06 — APPROVED** [review]

**Pull request merged**